### PR TITLE
GAME-16 Create registration logic

### PR DESCRIPTION
**Pull request by Birek100**
Create new Express.js route /register with layout containing registering form

### CODE REVIEW

File: views/errlogin.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<html> 
+<head> 
+	<link rel="stylesheet" type="text/css" href="/static/styles.css">
+	<title>Register</title> 
+</head> 
+
+<body> 
+<h1>Error. Please try one more time.</h1>
+
```
Comment (oskarszura) [inline views/errlogin.html]: No, no... format it nicely! no unnecessary blank lines / every tag should have proper indentation.

File: index.js [hunk 2]
```diff
@@ -5,16 +5,50 @@ const fs = require('fs');
...
+			res.send(errloginHTML);
+			return;
+		}
+
+	}
+
+	// SET COOKIE
+});
+
+
```
Comment (oskarszura) [inline index.js]: No unnecessary blank lines

File: index.js [hunk 3]
```diff
@@ -5,16 +5,50 @@ const fs = require('fs');
 const loginHTML = fs.readFileSync('./views/login.html','utf8'); 
 const gameHTML = fs.readFileSync('./views/game.html','utf8');
 const registerHTML = fs.readFileSync('./views/register.html','utf8');
-app.use(express.urlencoded())
+const errloginHTML = fs.readFileSync('./views/errlogin.html', 'utf8');
+app.use(express.urlencoded());
```
Comment (oskarszura) [inline index.js]: Add new line here, blocks of code should be logically separated.

File: index.js [hunk 4]
```diff
@@ -5,16 +5,50 @@ const fs = require('fs');
...
-})
+  fs.appendFileSync('users.db', email + ':' + password +'\n', {flags: 'a+'});
+  res.end()
+});
+
+app.get('/login-auth', (req, res) => {
+	const name = req.query.name;
+	const pass = req.query.password;
+
+	console.log(req.query.name)
```
Comment (oskarszura) [inline index.js]: Remove console logs - you should use them only while debugging.

File: index.js [hunk 5]
```diff
@@ -5,16 +5,50 @@ const fs = require('fs');
...
+			return;
+		}
+		else {
+			res.send(errloginHTML);
+			return;
+		}
+
+	}
+
+	// SET COOKIE
```
Comment (oskarszura) [inline index.js]: This comment is redudant? Please remove it.

File: index.js [hunk 6]
```diff
@@ -5,8 +5,50 @@ const fs = require('fs');
...
 app.use('/static', express.static('public'));
 app.get('/', (req, res) => res.send(loginHTML))
 app.get('/game', (req, res) => res.send(gameHTML))
 app.get('/register', (req, res) => res.send(registerHTML))
+app.get('/errlogin', (req, res) => res.send(errloginHTML))
+app.post('/register', (req, res) => {
+  const email = req.body.email
+  const password = req.body.password
+  fs.appendFileSync('users.db', email + ':' + password +'\n', {flags: 'a+'});
+  res.end()
```
Comment (oskarszura) [inline index.js]: Use colons everywhere!

File: index.js (hunk 4 above)
Comment (oskarszura) [reply index.js]: you still didn't remove console logs

File: index.js [hunk 7]
```diff
@@ -2,11 +2,53 @@ const express = require('express')
...
+	console.log(req.query.password)
+	const registeredUsers = fs.readFileSync('users.db').toString().split("\n");
+	console.log(registeredUsers)
+
+	for(let i = 0; i < registeredUsers.length; i++)
+	{
+		const user = registeredUsers[i];
+		const n = user.split(':')[0];
+		const p = user.split(':')[1];
+
```
Comment (oskarszura) [inline index.js]: Why so many blank new lines?

File: index.js [hunk 8]
```diff
@@ -2,11 +2,53 @@ const express = require('express')
...
+		const n = user.split(':')[0];
+		const p = user.split(':')[1];
+		
+
+		if (n === name && p === pass) {
+			res.cookie('sid', n + p, { maxAge: 900000, httpOnly: true });
+			res.send(gameHTML);
+			return;
+		}
+
```
Comment (oskarszura) [inline index.js]: Why space bettween `if` and `else`?

File: index.js [hunk 9]
```diff
@@ -2,11 +2,53 @@ const express = require('express')
...
+		if (n === name && p === pass) {
+			res.cookie('sid', n + p, { maxAge: 900000, httpOnly: true });
+			res.send(gameHTML);
+			return;
+		}
+		
+		else {
+			res.send(errloginHTML);
+			return;
```
Comment (oskarszura) [inline index.js]: Do you need `return` here?

File: views/errlogin.html [hunk 10]
```diff
@@ -0,0 +1,13 @@
+<html> 
+<head> 
+	<link rel="stylesheet" type="text/css" href="/static/styles.css">
+	<title>Register</title> 
+</head> 
+
```
Comment (oskarszura) [inline views/errlogin.html]: Why blank space here?

File: views/errlogin.html [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+<body> 
+<h1>Error. Please try one more time.</h1>
+
+
+</body> 
+
```
Comment (oskarszura) [inline views/errlogin.html]: Why blank space here?

File: views/login.html [hunk 12]
```diff
@@ -1,31 +1,39 @@
 <html>
+
```
Comment (oskarszura) [inline views/login.html]: Why blank space here?

File: views/login.html [hunk 13]
```diff
@@ -1,31 +1,39 @@
...
 <head>
 	<link rel="stylesheet" type="text/css" href="/static/styles.css">
 	<script src="/static/scripts.js"></script>
 	<title>Penguin Game</title>
 </head>
+
```
Comment (oskarszura) [inline views/login.html]: Why blank space here?

File: views/login.html [hunk 14]
```diff
@@ -1,31 +1,39 @@
...
 <body>
 	<div class="container">
+
```
Comment (oskarszura) [inline views/login.html]: Why blank space here?

File: views/login.html [hunk 15]
```diff
@@ -1,31 +1,39 @@
...
 	<title>Penguin Game</title>
 </head>
+
 <body>
 	<div class="container">
+		
 		<header>
 			<h1>Penguin Game</h1>
 		</header>
+
```
Comment (oskarszura) [inline views/login.html]: Why blank space here?

File: views/login.html [hunk 16]
```diff
@@ -1,31 +1,39 @@
...
 			<label for="password">Password</label><br>
-			<input id="password" type="password" required>
+			<input name="password" type="password" required>
 		</div>
 		<div class="register">
 			<input type="submit" value="Play">
 		</div>
+	</form>
 	</section>
+
```
Comment (oskarszura) [inline views/login.html]: Why blank space here?

File: views/login.html [hunk 17]
```diff
@@ -1,31 +1,39 @@
...
 		<div class="register">
 			<input type="submit" value="Play">
 		</div>
+	</form>
 	</section>
+	
 	<footer>
 		All rights reserved &copy
 	</footer>
+
```
Comment (oskarszura) [inline views/login.html]: Why blank space here?

**Comment by Birek100** [comment]
Now is ok ??

**Comment by oskarszura** [comment]
🚢 🚀 👍 - merge it!

**Comment by oskarszura** [comment]
just one thing 
> GAME-16 Create registration logic 
into
> GAME-16 **-** Create registration logic 

- please be consistent on the naming things.

**Pull request merged**